sources/api/pluto/src/main.rs

```diff
@@ -146,23 +125,22 @@ use error::PlutoError;
 
 type Result<T> = std::result::Result<T, PlutoError>;
 
-async fn get_text_from_imds(client: &Client, uri: &str, session_token: &str) -> Result<String> {
-    client
-        .get(uri)
-        .header("X-aws-ec2-metadata-token", session_token)
-        .send()
-        .await
-        .context(error::ImdsRequest { method: "GET", uri })?
-        .error_for_status()
-        .context(error::ImdsResponse { uri })?
-        .text()
+async fn get_max_pods(imdsclient: &mut ImdsClient) -> Result<String> {
+    let instance_type_target = "instance-type";
+    let instance_type = match imdsclient
+        .fetch_metadata(&instance_type_target, "instance type")
         .await
-        .context(error::ImdsText { uri })
-}
-
-async fn get_max_pods(client: &Client, session_token: &str) -> Result<String> {
-    let instance_type =
-        get_text_from_imds(&client, IMDS_INSTANCE_TYPE_ENDPOINT, session_token).await?;
+    {
+        Ok(Some(instance_type)) => instance_type,
+        Ok(None) => {
+            return error::ImdsData {
+                target: instance_type_target,
+                target_type: "meta-data",
+            }
+            .fail()
+        }
+        Err(e) => Err(e).context(error::ImdsRequest)?,
+    };
```

Comment:
I'm pretty sure this can be simplified a bit.
Can you try `let instance_type = imdsclient
        .fetch_metadata(&instance_type_target, "instance type").context(error::ImdsRequest)?.context(error::ImdsData{target : instance_type_target, target_type: "meta-data"})?;`

Same for all other similar constructs in the PR.